Spigot-Server-Patches/0449-Per-World-Spawn-limits.patch

```diff
@@ -0,0 +1,49 @@
+From 7136ff814df7a69967d3957a4c613848752aa597 Mon Sep 17 00:00:00 2001
+From: chase <chasewhip20@gmail.com>
+Date: Sun, 15 Mar 2020 18:32:22 -0600
+Subject: [PATCH] Per World Spawn limits
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 7ca67a4a..e3262642 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -668,4 +668,18 @@ public class PaperWorldConfig {
+     private void zombieVillagerInfectionChance() {
+         zombieVillagerInfectionChance = getDouble("zombie-villager-infection-chance", zombieVillagerInfectionChance);
+     }
++
++    public int spawnLimitMonsters;
++    public int spawnLimitAnimals;
++    public int spawnLimitWaterAnimals;
++    public int spawnLimitAmbient;
++
++    private void perWorldSpawnLimit(){
++        String path = "spawn-limits";
++
++        spawnLimitMonsters = getInt(path + ".monsters", -1);
++        spawnLimitAnimals = getInt(path + ".animals", -1);
++        spawnLimitWaterAnimals = getInt(path + ".water-animals", -1);
++        spawnLimitAmbient = getInt(path + ".ambient", -1);
++    }
+ }
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index 1a5ee341..5851606f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -324,6 +324,12 @@ public class CraftWorld implements World {
+         this.generator = gen;
+ 
+         environment = env;
++        //Paper start - per world spawn limits
++        monsterSpawn = world.paperConfig.spawnLimitMonsters;
++        animalSpawn = world.paperConfig.spawnLimitAnimals;
++        waterAnimalSpawn = world.paperConfig.spawnLimitWaterAnimals;
++        ambientSpawn = world.paperConfig.spawnLimitAmbient;
++        //Paper end - per world spawn limits
```

Comment:
end doesn't really need the comment (and removing it would be more in line to how it's done everywhere else)

Reply:
this isn't an issue that needs commenting

Reply:
Kenny, I want to let you know about the time wasted from rebuilding patches.